src/toil/common.py

```diff
@@ -804,15 +804,9 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                         self.config.clean == "always"):
 
                 try:          
-                    name, _ = self.parseLocator(self.config.jobStore)
-
-                    if name == 'file':
-                        jobStoreName = self._jobStore.jobStoreDir                  
-                    else:
-                        jobStoreName = self._jobStore.locator
-                        
-                    status = ToilStatus.getStatus(jobStoreName)
-
+                    _, path = self.parseLocator(self.config.jobStore)
+                    status = ToilStatus.getStatus(path)
```

Comment:
Are you sure you don't just want to always pass the whole locator through here? `getStatus` needs all the same info about how to contact the job store that Toil in general does.